admin/category_product_listing.php

```diff
@@ -25,1230 +25,1656 @@
-                require DIR_WS_MODULES . $zc_products->get_handler($product_type) . '/delete_product_confirm.php';
-              }
-              //--------------PRODUCT_TYPE_SPECIFIC_INSTRUCTIONS_GO__ABOVE__HERE--------------------------------------------------------
-              // now do regular non-type-specific delete:
-              // remove product from all its categories:
-              for ($k = 0, $m = count($product_categories); $k < $m; $k++) {
-                  zen_unlink_product_from_category($product_id, $product_categories[$k]);
-              }
-              // confirm that product is no longer linked to any categories
-              $count_categories = zen_get_linked_categories_for_product($product_id);
-              // if not linked to any categories, do delete:
-              if (empty($count_categories)) {
-                zen_remove_product($product_id, $delete_linked);
-              }
-            } // endif $do_delete_flag
-            // if this is a single-product delete, redirect to categories page
-            // if not, then this file was called by the cascading delete initiated by the category-delete process
-            if ($action === 'delete_product_confirm') {
-              zen_redirect(zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'cPath=' . $cPath));
+            if (isset($_POST['flag']) && ($_POST['flag'] === '0' || $_POST['flag'] === '1')) {
+                if (isset($_GET['pID'])) {
+                    zen_set_product_status($_GET['pID'], $_POST['flag']);
+                }
             }
-          }
 
-          zen_remove_category($categories[$i]['id']);
-        } // end for loop
-      }
-      zen_redirect(zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'cPath=' . $cPath));
-      break;
-    case 'move_category_confirm':
-      if (isset($_POST['categories_id']) && ($_POST['categories_id'] !== (int)$_POST['move_to_category_id'])) {
-        $categories_id = zen_db_prepare_input($_POST['categories_id']);
-        $new_parent_id = zen_db_prepare_input($_POST['move_to_category_id']);
+            zen_redirect(
+                zen_href_link(FILENAME_CATEGORY_PRODUCT_LISTING, 'cPath=' . $_GET['cPath'] . '&pID=' . $_GET['pID'] . (isset($_GET['page']) ? '&page=' . $_GET['page'] : '') . $search_parameter)
+            );
+            break;
+        case 'delete_category_confirm':
 
-        $path = explode('_', zen_get_generated_category_path_ids($new_parent_id));
+            // future cat specific deletion
+            $delete_linked = 'true';
+            if (isset($_POST['delete_linked']) && $_POST['delete_linked'] !== '') {
+                $delete_linked = $_POST['delete_linked'];
+            }
+
+            // delete category and products
+            if (!empty($_POST['categories_id']) && $_POST['categories_id'] !== (int)TOPMOST_CATEGORY_PARENT_ID) {
+                $categories_id = zen_db_prepare_input($_POST['categories_id']);
```

Comment:
zen_db_prepare_input not needed, variable is pre-sanitized to an int.

Reply:
ok, next PR.